### ISSUE TITLE
Fixed dimensions (width and height) of AbstractScrollArea

### ISSUE THREAD

**Issue by mever** [qooxdoo/qooxdoo#9351]
Hi,

When changing a custom widget to support scrolling one can choose to inherit from {qx.ui.core.scroll.AbstractScrollArea} and place the children on the 'pane' instead. This adds scroll capabilities without too much hassle.

There is a catch, however. AbstractScrollArea hard codes the preferred width and height. Making the described change not enough for a drop-in replacement. To work around this one can set the width and height to null in the constructor of the custom widget. Although this solves the issue many people will stumble with this non-intuitive behavior.

This issue is meant to help people and may inspire a change in the defaults. Although the latter may break backwards compatibility (as some implementations may rely on the fixed width and height). It makes Qooxdoo even more of a pleasure to work with.

Cheers,
Martijn

**Comment by level420** [comment]
@mever thank you for bringing this up. 
Note that you also could override (via refine) the `height` and `width` property with `null`.

**Comment by level420** [comment]
@mever if you like to contribute, you could add some documentation to the `qx.ui.core.scroll.AbstractScrollArea` class or you could create an example within the demo browser ( http://www.qooxdoo.org/current/demobrowser/# ) which will for sure help others searching for a similar solution as you did.

**Comment by mever** [comment]
How about removing the refined properties from the AbstractScrollArea and putting it as minor BC break for 6.0? I find it hard to imaging users rely on the exact dimensions of 100x100 in their application. They probably set their own dimensions. A minor price to pay for more intuitive components don't you think?

**Comment by level420** [comment]
Sure! Please create a PR for this. Thank you!

**Comment by mever** [comment]
I use Intellij Webstorm which understands JSDoc (http://usejsdoc.org/index.html). The 'jsdoc' branch introduces some changes that make my IDE help to understand the types used throughout the code. The problem is that the generator isn't able to parse these. It says:
```
      - Warning: qx.test.ui.core.AbstractScrollArea (30): Unable to parse JSDoc entry: @type qx.test.ui.core.AbstractScrollArea.fixture.CustomWidget
      - Warning: qx.test.ui.core.AbstractScrollArea (33): Unable to parse JSDoc entry: @type qx.ui.container.Composite
      - Warning: qx.test.ui.core.AbstractScrollArea (43): Unable to parse JSDoc entry: @param {qx.ui.core.LayoutItem} widget
      - Warning: qx.test.ui.core.AbstractScrollArea (50): Unable to parse JSDoc entry: @param {string} side Either 'x' or 'y'.
      - Warning: qx.test.ui.core.AbstractScrollArea (79): Unable to parse JSDoc entry: @param {qx.ui.core.Widget|Bounds} expected
      - Warning: qx.test.ui.core.AbstractScrollArea (80): Unable to parse JSDoc entry: @param {qx.ui.core.Widget} actual

```

These are valid JSDoc annotations, but I also know Qooxdoo was using annotations around the same time JSDoc annotations where introduced. It is unfortunate Qooxdoo and JSDoc are not totally compatible.

Do you think it is a good idea to open an issue for changing the generator to be able to parse JSDoc comments from the 'official' (http://usejsdoc.org/index.html) specification?

**Comment by mever** [comment]
@level420 can this issue be closed? #9355 is merged now.

**Comment by level420** [comment]
@mever sure. fixed via #9355
